Mathlib/Combinatorics/SimpleGraph/Matching.lean

```diff
@@ -133,6 +133,53 @@ lemma IsMatching.coeSubgraph {G' : Subgraph G} {M : Subgraph G'.coe} (hM : M.IsM
   · obtain ⟨_, hw', hvw⟩ := (coeSubgraph_adj _ _ _).mp hy
     rw [← hw.2 ⟨y, hw'⟩ hvw]
 
+lemma IsMatching.exists_of_disjoint_sets_of_equiv {s t : Set V} (h : Disjoint s t)
+    (f : s ≃ t) (hadj : ∀ v : s, G.Adj v (f v)) :
+    ∃ M : Subgraph G, M.verts = s ∪ t ∧ M.IsMatching := by
+  haveI (v : V) : Decidable (v ∈ s) := Classical.dec _
+  use {
+    verts := s ∪ t
+    Adj := fun v w ↦ (if h : v ∈ s then f ⟨v, h⟩ = w else False) ∨
+          (if h : w ∈ s then f ⟨w, h⟩ = v else False)
+    adj_sub := by
+      intro v w h
+      simp only [dite_else_false] at h
+      cases' h with hl hr
+      · rw [← hl.choose_spec]
+        exact hadj ⟨v, hl.choose⟩
+      · rw [← hr.choose_spec, G.adj_comm]
+        exact hadj ⟨w, hr.choose⟩
+    edge_vert := by
+      intro v w hvw
+      simp only [dite_else_false] at hvw
+      cases' hvw with hl hr
+      · exact Set.mem_union_left t hl.choose
+      · rw [← hr.choose_spec]
+        exact Set.mem_union_right s (f _).coe_prop
+  }
+
+  simp only [dite_else_false, true_and]
+  intro v hv
+  simp only [Set.mem_union] at hv
```

Comment:
```suggestion
  simp only [Subgraph.IsMatching, Set.mem_union, true_and]
  intro v hv
```
Your `intro` line is unfolding the `Subgraph.IsMatching` definition, in order to use Set.mem_union. But if I ask `simp` to unfold that definition instead, then I can combine the two `simp`-calls into one (and the new proof should be a bit more robust if `Subgraph.IsMatching` changes definition, for whatever reason)